### ISSUE TITLE
Manual score creations

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#2275]
We've had the need in the past to manually create scores on the platform based on paper scoring from judges, sometimes after scoring is otherwise closed. Right now it takes a developer to do it behind the scenes; admins should have this ability instead.

**Comment by rsgonzal** [comment]
Not great to do database surgery. Would be good for admin to have power but need to think about implications

**Comment by rsgonzal** [comment]
Another option is to not close judging until all the scores have been reviewed

**Comment by stenington** [comment]
True, although we potentially face a race condition where scores keep coming in as we keep trying to review and get a handle on the overall picture. Not sure this is super likely, I know we more often struggle to get enough scores, but just pointing out a potential weakness in this approach for consideration.

**Comment by stenington** [comment]
If we wanted to give admins the ability to enter scores through the admin, I think we'd need to prototype that out and first come up with a UI admins felt was workable, and then break that down and estimate the work. I don't believe the admin currently lets a user create anything from scratch like this would, so there's not a current pattern I could follow and adapt for this usage. I think designing out a solution here is totally doable, but ups the effort score significantly. 

There's a potential halfway solution though. I found a utility in the codebase that would import a score from a CSV file. We could spend some time dusting that off and making sure it works correctly and/or updating it to work correctly, and then we could provide admins with a CSV template that they would fill out, that a developer would then need to run the tool on to create the score. But, running a well-established and tested tool is less time-consuming than winging it, which is why I think this is potentially a viable halfway solution to the stated goal of this ticket.

So I think next steps are either:
1. start designing the admin screens for score entry, or
2. assess the state of the score import tool

depending on which route sounds better.

**Comment by stenington** [comment]
#2362 opened for option 2

**Comment by stenington** [comment]
Option 2 did not pan out, the old score import tool has serious limitations that make a manual process preferable. Reviving this ticket.

**Comment by hellafitz** [comment]
- test documentation for reproducibility by Shaun
- determine approx threshold for the number of scores we'd feel comfortable doing manually (vs needing a more scaled solution) e.g. 200 is clearly too many [context: 2018/19 total was 3 scores]

**Comment by hellafitz** [comment]
Mike: it'd be helpful if 
- Admins can manually assign judges to submissions in some cases
- and if judges could do the judging of those assignments after we've closed submissions

Add a enter manual score option [confounding factor, certain amount of judging logic is embedded in a Vue app, so would likely require a certain amount of duplication outside of Vue on the dashboard]

**Comment by hellafitz** [comment]
per Amy, the estimated range for how many manual scores have needed to be created in the past is:
- Historically, never above 100
- Average of 20-50
- 5-10 sometimes added at end

**Comment by shaunxp20** [comment]
> Another option is to not close judging until all the scores have been reviewed

I think this is an option worth exploring further. 🤔 😄 

Seems like reviewing scores and adding new ones as necessary **before** the judging window closes is a better way to go about it than adding them afterwards (either via database surgery or an admin gui; but of these two options an admin gui would definitely be better). And we have existing functionality to assign a submission to a particular judge, and have functionality to delete scores via the admin gui (if we get/have too many scores). #just-a-thought! 😃
